### PR TITLE
[zh-cn] Init translation of 'vue refs focus management'

### PR DESCRIPTION
**Pull request by jasonren0403**
### Description

reduce sidebar flaw by one :(

### CODE REVIEW

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/vue_refs_focus_management/index.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+
+为了给用户一个更好的体验，我们将添加代码来控制焦点，以便在编辑表单显示时将其设置为编辑字段。当用户取消或保存他们的编辑时，我们也要把焦点放回“Edit”按钮上。为了设置焦点，我们需要多了解一点 Vue 的内部工作方式。
+
+## 虚拟 DOM 和模板引用
+
+Vue 和其他一些框架一样，使用一个虚拟 DOM（VDOM）来管理元素。这意味着 Vue 在内存中保留了我们应用程序中所有节点的表示。任何更新首先在内存中的节点上执行，然后所有需要对页面上的实际节点进行的更改都会被批量同步。
+
+由于读写实际的 DOM 节点通常比虚拟节点更昂贵，这可以带来更好的性能。然而，这也意味着在使用框架时，你往往不应该直接通过本地浏览器 API（如 [`Document.getElementById`](/zh-CN/docs/Web/API/Document/getElementById)）来编辑你的 HTML 元素，因为这会导致 VDOM 和真实 DOM 不同步。
+
+相反，如果你需要访问底层DOM节点（比如设置焦点时），你可以使用 [Vue 模板引用](https://cn.vuejs.org/guide/essentials/template-refs.html)。对于自定义的 Vue 组件，你也可以使用 refs 来直接访问子组件的内部结构，然而这样做应该是谨慎的，因为这会使代码更难推理和理解。
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/vue_refs_focus_management/index.md:61]:
```suggestion
相反，如果你需要访问底层 DOM 节点（比如设置焦点时），你可以使用 [Vue 模板引用](https://cn.vuejs.org/guide/essentials/template-refs.html)。对于自定义的 Vue 组件，你也可以使用 refs 来直接访问子组件的内部结构，然而这样做应该是谨慎的，因为这会使代码更难推理和理解。
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/vue_refs_focus_management/index.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+
+这个行为可能令人震惊。此外，当再次按下 <kbd>Tab</kbd> 时会发生什么，取决于你所使用的浏览器。同样地，如果你保存或取消了你的编辑，当你移回非编辑视图时，焦点会再次消失。
+
+为了给用户一个更好的体验，我们将添加代码来控制焦点，以便在编辑表单显示时将其设置为编辑字段。当用户取消或保存他们的编辑时，我们也要把焦点放回“Edit”按钮上。为了设置焦点，我们需要多了解一点 Vue 的内部工作方式。
+
+## 虚拟 DOM 和模板引用
+
+Vue 和其他一些框架一样，使用一个虚拟 DOM（VDOM）来管理元素。这意味着 Vue 在内存中保留了我们应用程序中所有节点的表示。任何更新首先在内存中的节点上执行，然后所有需要对页面上的实际节点进行的更改都会被批量同步。
+
+由于读写实际的 DOM 节点通常比虚拟节点更昂贵，这可以带来更好的性能。然而，这也意味着在使用框架时，你往往不应该直接通过本地浏览器 API（如 [`Document.getElementById`](/zh-CN/docs/Web/API/Document/getElementById)）来编辑你的 HTML 元素，因为这会导致 VDOM 和真实 DOM 不同步。
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/vue_refs_focus_management/index.md:59]:
```suggestion
由于读写实际的 DOM 节点通常比虚拟节点开销更大，虚拟 DOM 可以带来更好的性能。然而，这也意味着在使用框架时，你往往不应该直接通过本地浏览器 API（如 [`Document.getElementById`](/zh-CN/docs/Web/API/Document/getElementById)）来编辑你的 HTML 元素，因为这会导致 VDOM 和真实 DOM 不同步。
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/vue_refs_focus_management/index.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+```html
+<input
+  :id="id"
+  ref="labelInput"
+  type="text"
+  autocomplete="off"
+  v-model.lazy.trim="newName" />
+```
+
+接下来，在你的组件对象中添加一个 `mounted()` 属性——**注意，这不应该放在 `methods` 属性里面，而应该与 `props'、`data()` 和 `methods` 处于同一个层次。**生命周期方法是独立的特殊方法，不应与用户定义方法放在一起。这些方法不接受参数。注意，你不能在这里使用箭头函数，因为我们需要访问 `this` 来访问我们的 `labelInput`。
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/vue_refs_focus_management/index.md:173]:
```suggestion
接下来，在你的组件对象中添加一个 `mounted()` 属性——**注意，这不应该放在 `methods` 属性里面，而应该与 `props`、`data()` 和 `methods` 处于同一个层次。**生命周期方法是独立的特殊方法，不应与用户定义方法放在一起。这些方法不接受参数。注意，你不能在这里使用箭头函数，因为我们需要访问 `this` 来访问我们的 `labelInput`。
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/vue_refs_focus_management/index.md [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+3. `beforeMount()`：在你的模板被编译后，但在你的组件被渲染到实际的 DOM 前运行。
+4. `mounted()`：在你的组件被安装到 DOM 后运行。可以在这里访问 `refs`。
+5. `beforeUpdate()`：当你的组件中的数据发生变化时，但在变化被渲染到 DOM 之前，就会运行。
+6. `updated()`：只要你的组件中的数据发生变化，并且在变化被渲染到 DOM 之后，就会运行。
+7. `beforeDestroy()`：在一个组件从 DOM 中移除之前运行。
+8. `destroyed()`：在一个组件被从 DOM 中移除后运行。
+9. `activated()`：只用于用特殊的 `keep-alive` 标签包装的组件。在组件被激活后运行。
+10. `deactivated()`：只用于用特殊的 `keep-alive` 标签包装的组件。在组件被停止激活后运行。
+
+> **备注：** Vue文档提供了一个[很好的图表，可以直观地看到这些钩子何时发生](https://cn.vuejs.org/guide/essentials/lifecycle.html#lifecycle-diagram)。这篇文章来自 [Digital Ocean 社区博客，更深入地探讨了生命周期方法](https://www.digitalocean.com/community/tutorials/vuejs-component-lifecycle)。
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/vue_refs_focus_management/index.md:158]:
```suggestion
> **备注：** Vue 文档提供了一个[很好的图表，可以直观地看到这些钩子何时发生](https://cn.vuejs.org/guide/essentials/lifecycle.html#lifecycle-diagram)。这篇文章来自 [Digital Ocean 社区博客，更深入地探讨了生命周期方法](https://www.digitalocean.com/community/tutorials/vuejs-component-lifecycle)。
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/vue_refs_focus_management/index.md [hunk 5]
```diff
@@ -0,0 +1,228 @@
+---
+title: 使用 Vue 模板引用进行焦点管理
+slug: Learn/Tools_and_testing/Client-side_JavaScript_frameworks/Vue_refs_focus_management
+---
+
+{{LearnSidebar}}{{PreviousMenuNext("Learn/Tools_and_testing/Client-side_JavaScript_frameworks/Vue_conditional_rendering","Learn/Tools_and_testing/Client-side_JavaScript_frameworks/Vue_resources", "Learn/Tools_and_testing/Client-side_JavaScript_frameworks")}}
+
+使用 Vue 所做的事情快要结束了。最后一个需要讲述的内容是焦点管理，换句话说，也是如何提升我们应用的键盘无障碍性。我们将看看使用 **Vue refs** 来处理这个问题——这是一个进阶功能，允许你直接访问虚拟 DOM 之下的底层 DOM 节点，或者从一个组件直接访问一个子组件的内部 DOM 结构。
```
Comment (yin1999) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/vue_refs_focus_management/index.md:8]:
```suggestion
使用 Vue 所做的事情快要结束了。最后一个需要讲述的内容是焦点管理，换句话说，也是如何提升我们应用的键盘无障碍性。我们将看看使用 **Vue ref** 来处理这个问题——这是一个进阶功能，允许你直接访问虚拟 DOM 之下的底层 DOM 节点，或者从一个组件直接访问一个子组件的内部 DOM 结构。
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/vue_refs_focus_management/index.md [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+
+这个行为可能令人震惊。此外，当再次按下 <kbd>Tab</kbd> 时会发生什么，取决于你所使用的浏览器。同样地，如果你保存或取消了你的编辑，当你移回非编辑视图时，焦点会再次消失。
+
+为了给用户一个更好的体验，我们将添加代码来控制焦点，以便在编辑表单显示时将其设置为编辑字段。当用户取消或保存他们的编辑时，我们也要把焦点放回“Edit”按钮上。为了设置焦点，我们需要多了解一点 Vue 的内部工作方式。
+
+## 虚拟 DOM 和模板引用
+
+Vue 和其他一些框架一样，使用一个虚拟 DOM（VDOM）来管理元素。这意味着 Vue 在内存中保留了我们应用程序中所有节点的表示。任何更新首先在内存中的节点上执行，然后所有需要对页面上的实际节点进行的更改都会被批量同步。
+
+由于读写实际的 DOM 节点通常比虚拟节点开销更大，虚拟 DOM 可以带来更好的性能。然而，这也意味着在使用框架时，你往往不应该直接通过本地浏览器 API（如 [`Document.getElementById`](/zh-CN/docs/Web/API/Document/getElementById)）来编辑你的 HTML 元素，因为这会导致 VDOM 和真实 DOM 不同步。
```
Comment (yin1999) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/vue_refs_focus_management/index.md:59]:
```suggestion
由于读写实际的 DOM 节点通常比虚拟节点开销更大，虚拟 DOM 可以带来更好的性能。然而，这也意味着在使用框架时，你往往不应该直接通过原生浏览器 API（如 [`Document.getElementById`](/zh-CN/docs/Web/API/Document/getElementById)）来编辑你的 HTML 元素，因为这会导致 VDOM 和真实 DOM 不同步。
```

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/vue_refs_focus_management/index.md [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+
+相反，如果你需要访问底层 DOM 节点（比如设置焦点时），你可以使用 [Vue 模板引用](https://cn.vuejs.org/guide/essentials/template-refs.html)。对于自定义的 Vue 组件，你也可以使用 refs 来直接访问子组件的内部结构，然而这样做应该是谨慎的，因为这会使代码更难推理和理解。
+
+要在一个组件中使用 ref，你需要在你想访问的元素上添加一个 `ref` 属性，并为该属性的值添加一个字符串标识符。值得注意的是，在一个组件中，引用必须是唯一的。在同一时间呈现的两个元素不应该有相同的引用。
```
Comment (yin1999) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/vue_refs_focus_management/index.md:63]: coud we use "模板引用" for all those name, if it's not refering to a attribute :)

We may need to consider the translation for all the instances.

File: files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/vue_refs_focus_management/index.md [hunk 8]
```diff
@@ -0,0 +1,228 @@
+---
+title: 使用 Vue 模板引用进行焦点管理
+slug: Learn/Tools_and_testing/Client-side_JavaScript_frameworks/Vue_refs_focus_management
+---
+
+{{LearnSidebar}}{{PreviousMenuNext("Learn/Tools_and_testing/Client-side_JavaScript_frameworks/Vue_conditional_rendering","Learn/Tools_and_testing/Client-side_JavaScript_frameworks/Vue_resources", "Learn/Tools_and_testing/Client-side_JavaScript_frameworks")}}
+
+使用 Vue 所做的事情快要结束了。最后一个需要讲述的内容是焦点管理，换句话说，也是如何提升我们应用的键盘无障碍性。我们将看看如何使用 **Vue 模板引用** 来处理这个问题——这是一个进阶功能，允许你直接访问虚拟 DOM 之下的底层 DOM 节点，或者从一个组件直接访问一个子组件的内部 DOM 结构。
```
Comment (yin1999) [inline files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/vue_refs_focus_management/index.md:8]:
```suggestion
使用 Vue 所做的事情快要结束了。最后一个需要讲述的内容是焦点管理，换句话说，也是如何提升我们应用的键盘无障碍性。我们将看看如何使用 **Vue 模板引用**来处理这个问题——这是一个进阶功能，允许你直接访问虚拟 DOM 之下的底层 DOM 节点，或者从一个组件直接访问一个子组件的内部 DOM 结构。
```

**Review by yin1999 — APPROVED** [review]
Thanks

**Pull request merged**